dvc/repo/__init__.py

```diff
@@ -538,3 +539,29 @@ def func(out):
     def is_dvc_internal(self, path):
         path_parts = os.path.normpath(path).split(os.path.sep)
         return self.DVC_DIR in path_parts
+
+    def get_remote(self, remote=None):
```

Comment:
So do we need DataCloud any longer?

Reply:
Need to reimplement `push`/`pull` etc without it first.

Reply:
I suppose get_remote is just an incomplete copypaste of DataCloud._get_cloud?

Reply:
I left compat thing there, no point refactoring future dead code.

Reply:
Maybe you could use get_remote in DataCloud, so there is no copypasting?

Reply:
Looks like repo file/class has gotten too large (number of function per file, number of lines, etc)- check CC issues. 

Can create a ticket and PR to simplify discussion for this RFC? To keep this one simple and merge faster?

Unless I'm missing something and it's actually needed for the implementation.

Reply:
Actually, most of the "get_remote" can be a part of config, since most of it is just prioritizing things from config one over each other. I think this should be solved before merging, because having copypasted stuff from DataCloud here is not good.

Reply:
I don't think `.get_remote()` is a good part of config. This way we should move all instantiation to config making it a babylonian tower.

Reply:
I can continue removing DataCloud in this PR.

Reply:
Would be great to see some motivation behind this decision and some description of the suggested improvement. It's hard to review it w/o seeing where is it going.

Also, like I mentioned repo is already too complicated: `> 400 lines` + `> 20 methods`. There are Code Climate complaints about this change.